indico/util/marshmallow.py

```diff
@@ -163,3 +174,23 @@ def _deserialize(self, value, attr, data):
             }
         except ValueError as exc:
             raise ValidationError(unicode(exc))
+
+
+class HumanizedDate(Field):
+    """Marshmallow field for human-written dates used in REST APIs."""
+
+    def _serialize(self, value, attr, obj):
+        raise NotImplementedError
+
+    def _deserialize(self, value, attr, obj):
+        m = HUMANIZED_DATE_RE.match(value)
+        if not m:
+            raise ValidationError("Can't parse humanized date!")
+
+        if m.groupdict().get('iso_date'):
+            return parser.isoparse(value)
+        else:
+            unit = m.groupdict()['unit']
```

Comment:
Isn't there `.group('unit')` for that?

Reply:
I remember telling myself "I should definitely check the re API before I send the PR, since I'm pretty sure there was a nicer way of doing this". That clearly didn't happen 😆